crates/walrus-service/src/node/blob_event_processor.rs

```diff
@@ -316,12 +391,58 @@ impl BlobEventProcessor {
             )
             .inc();
 
+            // Increment the counter and create a guard that will decrement it when dropped
+            let current_pending_event_count = self.background_per_processor_pending_event_count
+                [processor_index]
+                .fetch_add(1, Ordering::SeqCst)
+                + 1;
+            walrus_utils::with_label!(
+                self.node
+                    .metrics
+                    .pending_processing_blob_event_in_background_processors,
+                &processor_index.to_string()
+            )
+            .set(i64::try_from(current_pending_event_count).unwrap_or(i64::MAX));
+
+            // Create the guard that will automatically decrement the counter when dropped
+            let guard = PendingEventGuard::new(
+                self.background_per_processor_pending_event_count[processor_index].clone(),
+                processor_index,
+                self.node.metrics.clone(),
+            );
+
+            // Send the wrapped event with the guard
+            let tracked_event = TrackedEvent {
+                event_handle,
+                blob_event,
+                _guard: guard,
+            };
+
             self.background_processor_senders[processor_index]
-                .send((event_handle, blob_event))
+                .send(tracked_event)
                 .map_err(|e| {
                     anyhow::anyhow!("failed to send event to background processor: {}", e)
```

Comment:
If this send fails (due to queue full), do we end up with higher metric counts because we are incrementing them before the send? Should we increment them after send

Reply:
The send cannot fail unless the receiver side quit. We are using an unbounded queue.

Also, send failure should terminate the process.